packages/forklift-console-plugin/src/modules/Overview/utils/helpers/OverviewUserSettings.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import React from 'react';
+
+import { loadFromLocalStorage, removeFromLocalStorage, saveToLocalStorage } from '@kubev2v/common';
+
+export interface OverviewUserSettings {
+  welcome?: WelcomeSettings;
+}
+
+interface WelcomeSettings {
+  hideWelcome: boolean;
+  save: (showWelcome: boolean) => void;
+  clear: () => void;
+}
+
+const parseOrClean = (key) => {
+  try {
+    return JSON.parse(loadFromLocalStorage(key)) ?? {};
+  } catch (e) {
+    removeFromLocalStorage(key);
+    console.error(`Removed invalid key [${key}] from local storage`);
+  }
+  return {};
+};
+
+const saveRestOrRemoveKey = (key: string, { rest }: { [k: string]: { [n: string]: unknown } }) => {
+  if (!Object.keys(rest).length) {
+    removeFromLocalStorage(key);
+  } else {
+    saveToLocalStorage(key, JSON.stringify({ ...rest }));
+  }
+};
+
+/**
+ * Deserialize user settings for the overview page - welcome card
+ *
+ * 1. user settings are stored in local storage as JSON encoded string
+ * 2. if data cannot be decoded it's removed from the local storage (auto clean-up)
+ *
+ * @param userSettingsKeySuffix - The key name together with PLUGIN_NAME used to load/save data.
+ */
+export const loadUserSettings = ({ userSettingsKeySuffix }): OverviewUserSettings => {
+  const key = `${process.env.PLUGIN_NAME}-${userSettingsKeySuffix}`;
+  const { hideWelcome } = parseOrClean(key);
+
+  return {
+    welcome: {
+      hideWelcome: typeof hideWelcome === 'boolean' ? hideWelcome : undefined,
+      save: (hideWelcome) =>
+        saveToLocalStorage(key, JSON.stringify({ ...parseOrClean(key), hideWelcome })),
+      clear: () => {
+        const { hideWelcome, ...rest } = parseOrClean(key);
+        saveRestOrRemoveKey(key, { hideWelcome, rest });
+      },
+    },
+  };
+};
+
+export const setWelcomeStateByUserSettings = (): [
+  userSettings: OverviewUserSettings,
+  hideWelcomeCard: boolean,
+  setHideWelcomeCard: React.Dispatch<React.SetStateAction<boolean>>,
+] => {
+  const [userSettings] = useState(() => loadUserSettings({ userSettingsKeySuffix: 'Overview' }));
+  const hideWelcomeCardInitState = userSettings?.welcome?.hideWelcome;
+  const [hideWelcomeCard, setHideWelcomeCard] = React.useState(hideWelcomeCardInitState);
+  return [userSettings, hideWelcomeCard, setHideWelcomeCard];
```

Comment:
hmm, this is strange,
if it's a hook for `userSettings`, then it should return `[userSettings, setUserSettings]`
if it's a hook for `hideWelcomeCard` it should return `[hideWelcomeCard, setHideWelcomeCard]`

it should not return both `userSettings` and `hideWelcomeCard`, is there a reason for return both in this case ?

Reply:
Done (this file was removed as part of the solution of merging the load settings handling into the context hook).